appendix-basic65.tex

```diff
@@ -9746,62 +9869,12 @@ \subsection{SYS}
 \item [Token:] \$9E
 \item [Format:] {\bf SYS} address [\{{\bf,} areg{\bf,} xreg{\bf,} yreg{\bf,}
 		zreg{\bf,} sreg\}]
-\item [Usage:]  Calls a machine language subroutine.
-                This can be a ROM-resident kernal routine
-                or any other routine which has previously been loaded or {\bf POKE}d
-                to RAM.
-
-               The CPU registers are loaded with the arguments
-               (if they're specified), then a subroutine call ({\bf JSR address})
-               is performed. {\bf JSR} is an assembly language instruction that is short
-               for {\bf J}ump to {\bf S}ub{\bf R}outine.
-               The called routine should exit with
-               an {\bf RTS} instruction. {\bf RTS} is another assembly
-               language instruction that is
-               short for {\bf R}e{\bf t}urn from {\bf S}ubroutine.
-               After the subroutine has returned,
-               the register contents will be saved,
-               and the execution of the BASIC program will
-               continue.
+\item [Usage:] Calls a machine language subroutine.
 
                {\bf address} start address of the subroutine.
-
-               If the address value is 16 bit (\$0000 - \$FFFF), the bank value,
-               that is currently valid (see {\bf BANK}) is examined. A
-               bank value of 128 lets the current mapping persist. That is:
-               RAM is only available at the address range (\$0000 - \$1FFF),
-               while BASIC ROM, KERNAL and I/O occupy the rest.
-               Short machine language programs may use the address range
-               (\$1800 - \$1FFF) which is only used by BASIC while a
-               graphics screen is open.
```

Comment:
Just curious why this was omitted, seems a handy reminder of the impact of BANK 128 and some advice for where to squeeze in a short assembly routine. Worthy of a new home?

Feel like I need to refresh my memory on SYS-behaviour with MAP and then re-read what you've got here. We've had some past discussions on discord about it, so that might help jog my memory.

Reply:
The updated description with regards to BANK is complete and more accurate. This section ends with a reference to the Memory chapter, which fully explains why SYS behaves the way it does.

I elided the factoid about $1800 because there's too much complex information here already. It's a useful fact but you need to know other things explained in the Memory chapter to understand why.